rpcs3/Emu/Cell/Modules/cellVoice.h

```diff
@@ -60,32 +45,139 @@ enum
-	CELLVOICE_PCM_INTEGER = 4,
+enum CellVoicePcmDataType
+{
+	CELLVOICE_PCM_NULL                  = -1,
+	CELLVOICE_PCM_FLOAT                 = 0,
+	CELLVOICE_PCM_FLOAT_LITTLE_ENDIAN   = 1,
+	CELLVOICE_PCM_SHORT                 = 2,
+	CELLVOICE_PCM_SHORT_LITTLE_ENDIAN   = 3,
+	CELLVOICE_PCM_INTEGER               = 4,
 	CELLVOICE_PCM_INTEGER_LITTLE_ENDIAN = 5,
+};
 
-	CELLVOICE_ATTR_ENERGY_LEVEL = 1000,
-	CELLVOICE_ATTR_VAD = 1001,
-	CELLVOICE_ATTR_DTX = 1002,
-	CELLVOICE_ATTR_AUTO_RESAMPLE = 1003,
-	CELLVOICE_ATTR_LATENCY = 1004,
+enum CellVoicePortAttr
+{
+	CELLVOICE_ATTR_ENERGY_LEVEL      = 1000,
+	CELLVOICE_ATTR_VAD               = 1001,
+	CELLVOICE_ATTR_DTX               = 1002,
+	CELLVOICE_ATTR_AUTO_RESAMPLE     = 1003,
+	CELLVOICE_ATTR_LATENCY           = 1004,
 	CELLVOICE_ATTR_SILENCE_THRESHOLD = 1005,
+};
 
-	CELLVOICE_APPTYPE_GAME_1MB = 1 << 29
+enum CellVoicePortState
+{
+	CELLVOICE_PORTSTATE_NULL      = -1,
+	CELLVOICE_PORTSTATE_IDLE      = 0,
+	CELLVOICE_PORTSTATE_READY     = 1,
+	CELLVOICE_PORTSTATE_BUFFERING = 2,
+	CELLVOICE_PORTSTATE_RUNNING   = 3,
+};
+
+enum CellVoicePortType
+{
+	CELLVOICE_PORTTYPE_NULL          = -1,
+	CELLVOICE_PORTTYPE_IN_MIC        = 0,
+	CELLVOICE_PORTTYPE_IN_PCMAUDIO   = 1,
+	CELLVOICE_PORTTYPE_IN_VOICE      = 2,
+	CELLVOICE_PORTTYPE_OUT_PCMAUDIO  = 3,
+	CELLVOICE_PORTTYPE_OUT_VOICE     = 4,
+	CELLVOICE_PORTTYPE_OUT_SECONDARY = 5,
+};
+
+enum CellVoiceSamplingRate
+{
+	CELLVOICE_SAMPLINGRATE_NULL  = -1,
+	CELLVOICE_SAMPLINGRATE_16000 = 16000,
+};
+
+enum CellVoiceVersionCheck
+{
+	CELLVOICE_VERSION_100 = 100
+};
+
+// Definitions
+enum
+{
+	CELLVOICE_MAX_IN_VOICE_PORT           = 32,
+	CELLVOICE_MAX_OUT_VOICE_PORT          = 4,
+	CELLVOICE_GAME_1MB_MAX_IN_VOICE_PORT  = 8,
+	CELLVOICE_GAME_1MB_MAX_OUT_VOICE_PORT = 2,
+	CELLVOICE_MAX_PORT                    = 128,
+	CELLVOICE_INVALID_PORT_ID             = 0xff,
+};
+
+struct CellVoiceBasePortInfo // aligned(64)
+{
+	CellVoicePortType portType; // TODO: correct type ?
+	CellVoicePortState state; // TODO: correct type ?
+	be_t<u16> numEdge;
+	vm::bptr<u32> pEdge;
+	be_t<u32> numByte;
+	be_t<u32> frameSize;
+};
+
+struct CellVoiceInitParam // aligned(32)
+{
+	CellVoiceEventType eventMask; // TODO: correct type ?
+	CellVoiceVersionCheck version; // TODO: correct type ?
+	be_t<s32> appType;
+	u8 reserved[32 - sizeof(s32) * 3];
+};
+
+struct CellVoicePCMFormat
+{
+	be_t<u8> numChannels;
+	be_t<u8> sampleAlignment;
+	CellVoicePcmDataType dataType; // TODO: correct type ?
+	CellVoiceSamplingRate sampleRate; // TODO: correct type ?
+};
+
+struct CellVoicePortParam // aligned(64)
+{
+	CellVoicePortType portType; // TODO: correct type ?
+	be_t<u16> threshold;
+	be_t<u16> bMute;
+	be_t<f32> volume;
+	union // TODO: what's this supposed to look like in rpcs3 ?
+	{
+		struct
+		{
+			CellVoiceBitRate bitrate;
+		} voice;
+		struct
+		{
+			u32 bufSize;
+			CellVoicePCMFormat format;
+		} pcmaudio;
+		struct
+		{
+			u32 playerId;
+		} device;
+	};
```

Comment:
add be_t<> to bitrate, format and playerId.

Reply:
bufSize not?

Reply:
yes I meant bufSize, not format